sdk/src/types/block/mana/mod.rs

```diff
@@ -65,27 +64,43 @@ impl ManaAllotments {
     }
 }
 
-fn verify_allotments<const VERIFY: bool>(allotments: &[ManaAllotment], _visitor: &()) -> Result<(), Error> {
+fn verify_allotments<const VERIFY: bool>(
+    allotments: &[ManaAllotment],
+    protocol_params: &ProtocolParameters,
+) -> Result<(), Error> {
     if VERIFY {
-        if !is_unique_sorted(allotments.iter()) {
-            return Err(Error::ManaAllotmentsNotUniqueSorted);
-        }
-        verify_allotments_sum(allotments)?;
+        verify_allotments_unique_sorted(allotments)?;
+        verify_allotments_sum(allotments, protocol_params)?;
     }
 
     Ok(())
 }
 
-fn verify_allotments_sum<'a>(allotments: impl IntoIterator<Item = &'a ManaAllotment>) -> Result<(), Error> {
+fn verify_allotments_unique_sorted<'a>(allotments: impl IntoIterator<Item = &'a ManaAllotment>) -> Result<(), Error> {
+    if !is_unique_sorted(allotments.into_iter()) {
+        return Err(Error::ManaAllotmentsNotUniqueSorted);
+    }
+    Ok(())
+}
+
+pub(crate) fn verify_allotments_sum<'a>(
```

Comment:
I forgot to rename these, do you ming adding mana please. Even more important if they are not private anymore

Reply:
sure